### PR TITLE
Issue 1209 Adds bulk edit tags modal to deployments page

### PR DESCRIPTION
**Pull request by MTrost**

### CODE REVIEW

File: code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+            [sixsq.nuvla.ui.utils.semantic-ui :as ui]
+            [sixsq.nuvla.ui.utils.semantic-ui-extensions :as uix]
+            [sixsq.nuvla.ui.messages.events :as messages-events]))
+
+
+(def modal-tags-set-id ::tags-set)
+(def modal-tags-add-id ::tags-add)
+(def modal-tags-remove-id ::tags-remove)
+(def modal-tags-remove-all ::tags-remove-all)
+(def tags-modal-ids [modal-tags-add-id modal-tags-set-id modal-tags-remove-id modal-tags-remove-all])
```
Comment (0xbase12) [inline code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs:21]: you use `set(tags-modal-ids)` quite often, why not defining a 
`(def tags-modal-ids-set (set tags-modal-ids))`

File: code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+              [:div [:a {:style {:cursor :pointer}
+                         :target :_blank
+                         :on-click
+                         (fn []
+                           (dispatch
+                             [::events/store-filter-and-open-in-new-tab
+                              (str/join " or "
+                                (map #(str "id='" % "'")
+                                  (->> @view-only-items :resources (map :id))))]))}
+                     (str "Open " (if (= not-editable 1) "it" "them") " in a new tab")]]])]
```
Comment (0xbase12) [inline code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs:249]: i18n

File: code/src/cljs/sixsq/nuvla/ui/edges/events.cljs [hunk 3]
```diff
@@ -226,11 +141,11 @@
       (dispatch [::messages-events/add
                  (let [{:keys [status message]} (response/parse-ex-info nuvlaboxes)]
                    {:header  (cond-> (str "failure getting nuvlaboxes")
-                                     status (str " (" status ")"))
+                               status (str " (" status ")"))
```
Comment (0xbase12) [inline code/src/cljs/sixsq/nuvla/ui/edges/events.cljs:144]: :header nuvlaedges

File: code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                     :color    color
+                     :disabled disabled?
+                     :active   true
+                     :style    {:margin-left "2rem"}
+                     :on-click (fn [] (reset! mode :confirming))}
+          [uix/Icon {:style {:margin 0}
+                     :name "fa-check"}]]]
+        [:div
+         [:span "Are you sure? "]
+         [uix/Button {:text     (str "Yes, " text)
```
Comment (0xbase12) [inline code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs:111]: i18n

File: code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+  (fn [{db :db} [_ {:keys [resource-key updated-tags call-back-fn refetch-event
+                           text operation filter-fn
+                           db-path singular plural]}]]
+    {::cimi-api-fx/operation-bulk [resource-key
+                                   (fn [result]
+                                     (let [updated     (-> result :updated)
+                                           success-msg (str updated " " (if (= 1 updated) singular plural) " updated with operation: " text)]
+                                       (dispatch [::messages-events/add
+                                                  {:header  "Bulk edit operation successful"
+                                                   :content success-msg
```
Comment (0xbase12) [inline code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs:146]: i18n

File: code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+
+(reg-event-fx
+  ::open-modal
+  (fn [{db :db} [_ {:keys [resource-key modal-id db-path on-open-modal-event]}]]
+    (let [fx (when (and ((set tags-modal-ids) modal-id)
+                     (not ((set tags-modal-ids) (get-in db [::edit-mode db-path]))))
+               [[:dispatch [on-open-modal-event]]
+                [:dispatch [::fetch-tags resource-key]]])]
+      {:db (assoc-in db [::edit-mode db-path] modal-id)
+       :fx fx})))
```
Comment (0xbase12) [inline code/src/cljs/sixsq/nuvla/ui/utils/bulk_edit_tags_modal.cljs:52]: when :fx nil warning logged in browser console

File: code/src/cljs/sixsq/nuvla/ui/plugins/table.cljs
Comment (0xbase12) [inline code/src/cljs/sixsq/nuvla/ui/plugins/table.cljs:1]: can we change `{:position :sticky :top "40px" :z-index 998}` to `39px`.
When scrolling in deployment page we can see deployments rows between the two sticky bars

File: code/src/cljs/sixsq/nuvla/ui/deployments/events.cljs [hunk 7]
```diff
@@ -177,12 +182,29 @@
...
+        {:filter filter :select "id"}
+        #(dispatch [::set-deployments-without-edit-rights %])]})))
+
+(reg-event-fx
+  ::set-deployments-without-edit-rights
+  (fn [{:keys [db]} [_ nuvlaboxes]]
+    (if (instance? js/Error nuvlaboxes)
+      (dispatch [::messages-events/add
+                 (let [{:keys [status message]} (response/parse-ex-info nuvlaboxes)]
+                   {:header  (cond-> (str "failure getting nuvlaboxes")
```
[the remark this answers is not in the document]
Comment (MTrost) [reply code/src/cljs/sixsq/nuvla/ui/deployments/events.cljs:206]: Even worse: should be deployments

**Pull request merged**